src/jlgen.jl

```diff
@@ -74,27 +85,109 @@ function invalidate(cache::CodeCache, replaced::MethodInstance, max_world, depth
     end
 end
 
-const CI_CACHE = CodeCache()
+
+## method overrides
+
+@static if VERSION >= v"1.7-"
+
+# use an overlay method table
+
+Base.Experimental.@MethodTable(GLOBAL_METHOD_TABLE)
+
+else
+
+# use an overlay world -- a special world that contains all method overrides
+
+const GLOBAL_METHOD_TABLE = nothing
+
+const override_world = typemax(Csize_t) - 1
```

Comment:
Uuh nasty. What happens if we have AMDGPU and CUDA both overriding the same method?

Reply:
Mayhem. Don't import those packages at the same time, for now (do you rely on that already?). Should work on 1.7.

Reply:
AMDGPU and CUDA.jl might not be used jointly, but parasitic packages like Enzyme.jl will.

Reply:
I _guess_ we could use different world ages, but that's going to be finicky...

Reply:
xD what about this is not finicky?